bioconda_utils/lint_functions.py

```diff
@@ -288,17 +288,30 @@ def _pin(env_var, dep_name):
     Generates a linting function that checks to make sure `dep_name` is pinned
     to `env_var` using jinja templating.
     """
+    pin_pattern = re.compile(r"\{{\{{\s*{}\s*\}}\}}\*".format(env_var))
     def pin(recipe, meta, df):
         # Note that we can't parse the meta.yaml using a normal YAML parser if it
         # has jinja templating
+        in_requirements = False
+        in_run = False
         for line in open(os.path.join(recipe, 'meta.yaml')):
+            if line.startswith("requirements:"):
+                in_requirements = True
+            elif in_requirements and line.strip().startswith("run:"):
+                in_run = True
+            elif in_requirements and line.strip().startswith("build:"):
+                in_run = False
+            elif not line.startswith(" "):
```

Comment:
Should/could be `elif not line.startswith((" ", "#")):`, right? That way, not indented comments are also handled.

Reply:
Good point.